### PR TITLE
Make default config an argument default instead of using null check

### PR DESCRIPTION
**Pull request by Ericson2314**
This looks cleaner to me, but subtly changes semantics in that passing in `config = null;` doesn't trigger defaults but rather (usually) errors.

### CODE REVIEW

**Comment by zimbatm** [comment]
I prefer to keep simple expressions as default arguments. Otherwise it makes it hard to follow what's going on before even entering the function.

**Comment by Ericson2314** [comment]
That's fine; my preference here is pretty light. This is taken from @Mathnerd314's extra commits and my main motivation for splitting it out was to separate this bikeshed from the rest of the patch.

**Comment by Ericson2314** [comment]
Any tie-breaker vote for this?

@zimbatm I combined the redundant comments and hoisted the `let` keyword to the previous line. Besides being what I consider good style, this shrinks the space taken which might make the fallback expression easier to read at a glance.

**Comment by Ericson2314** [comment]
@zimbatm also keep in mind that this anticipates the pure-impure split in my cross compiling branch. I'd be able to make `impure.nix` just fallback arguments, basically.

**Comment by zimbatm** [comment]
Alright, it's better now :)

**Comment by Ericson2314** [comment]
Thanks!
